auth/src/main/java/com/firebase/ui/auth/ui/ActivityHelper.java

```diff
@@ -96,6 +96,16 @@ public FirebaseUser getCurrentUser() {
         return getFirebaseAuth().getCurrentUser();
     }
 
+    public Intent getMergeFailedIntent() {
+        FirebaseUser user = getCurrentUser();
+        if (mFlowParams.shouldLinkUser && user != null) {
+            if (user.isAnonymous()) user.delete();
```

Comment:
I'm deleting the anonymous user here because the expected usage of `getMergeFailedIntent()` is that a non anonymous user will be signed into immediately after this method is called. If I don't delete the anonymous users, there will be a bunch of them hanging around that will never be user again. AFAIK, you can't log back into an existing anonymous user so not deleting them would make a messy user base in the Firebase console.

What I would really like to do is to delete the anonymous user once I'm signed in, but I can't figure out how to do that because I keep getting references to the newly signed in user.

If you believe we should play it safe and not delete the anonymous user, that's fine with me but I had one last idea: maybe I could delete the anonymous user and if signing into the new user fails, I just create a new anonymous user and pass down the old anonymous user's id through the Intent. What do you think? It would probably just be easier to not delete the anonymous user, but again, I take issue with the messy Firebase console. Do you know if Firebase automatically delete anonymous users if they haven't logged in for a while? That would solve the issue.